key.js

```diff
@@ -1,277 +1,446 @@
+        })
+    }
 
-const startAnimation = () => {
-    timerElement.style.display = ``;
-    newSequence();
-    startButton.style.fontSize = `45px`
-    let seconds = 60;
-    timer = setInterval(() => {
-        seconds--;
-        timerElement.textContent = `${seconds} `;
-        if (!seconds) {
-            clearInterval(timer);
-        }
-    }, 1000)
-    timerElement.style.boxShadow = `0px 0px 4px 3px darkgray`;
-    startButton.style.transform = ``
-    h3.style.display = ``;
-    keyBoard.style.display = ``;
-    startButton.textContent = `HOME`;
-    startButton.style.transform = `translateX(600px)`
-    startButton.style.backgroundColor = `red`;
-    startButton.style.color = `white`;
-    document.body.style.backgroundColor = `whitesmoke`;
-    startButton.style.border = `5px solid darkred`
-    headings.textContent = `Baby Level`;
-    headings.style.color = "darkslategray";
-    levels();
-}
-startButton.addEventListener('click', () => {
-    buttonPress()
-    bool = !bool;
-    if (bool) {
-        startAnimation();
-    } else {
-        homeAnimation();
-    }
-})
+
+    const entireLogic = () => {
+        const arrowCount = document.querySelector(viewModule.DOMStrings.randomKeyArrows).length;
+        let index = 0;
+        // Key Interactions Logic : 
+        addEventListener('keyup', (e) => {
+            view.resetArrowStyles();
+            const keyPress = e.key;
+            const currentSpan = document.querySelector(viewModule.DOMStrings.randomKeyArrows)[index];
+            const currentSpanStyles = () => {
+                currentSpan.classList.add('color-gray', 'arrow-transition-to-gray', 'font-size-70');
+            }
+            const greenCurrentSpanColor = () => { currentSpan.style.color = 'green' }
+            const redCurrentSpanColor = () => { currentSpan.style.color = 'red' }
+
+            const allDirections = {
+                directionUp: currentSpan.innerHTML === `↑`,
+                directionDown: currentSpan.innerHTML === `↓`,
+                directionLeft: currentSpan.innerHTML === `←`,
+                directionRight: currentSpan.innerHTML === `→`,
+            }
+            // Successful Sounds : 
+            const successfulGreenSounds = {
+                ArrowUp: allDirections.directionUp && greenCurrentSpanColor(),
+                ArrowDown: allDirections.directionDown && greenCurrentSpanColor(),
+                ArrowLeft: allDirections.directionLeft && greenCurrentSpanColor(),
+                ArrowRight: allDirections.directionRight && greenCurrentSpanColor()
+            };
+            const successfulRedSounds = {
+                ArrowUp: allDirections.directionDown && redCurrentSpanColor(),
+                ArrowDown: allDirections.directionUp && redCurrentSpanColor(),
+                ArrowLeft: allDirections.directionLeft && redCurrentSpanColor(),
+                ArrowRight: allDirections.directionRight && redCurrentSpanColor()
+            };
```

Comment:
From the aforementioned logic, you can use the `arrowKeys` variable here:

```suggestion
            const getSuccessfulColoredSounds = (arr, colorCallback) => {
              return arr.reduce((soundsTillNow, currentArrow) => {...soundsTillNow, [`Arrow${currentArrow}`]: allDirections[`direction${currentArrow}`] && colorCallback()}, {});
            }
            const successfulGreenSounds = getSuccessfulColoredSounds(arrowKeys, isArrowColorGreen);
            const successfulRedSounds = getSuccessfulColoredSounds(arrowKeys, isArrowColorRed);
```